### PR TITLE
Improve parsing error messages

### LINKED ISSUE #2160: Confusing error message with colon inside an identifier and a missing type signature
For example:
```
let x: Int := 2 in x
```
gives an error
```
The declaration is missing a type signature:
  x: Int := 2 
```
This doesn't help when you don't know that there should be a space before `:`.

### LINKED ISSUE #2167: The special error message for confusing `:=` with `=` no longer works
Typechecking
```
module eq;

import Stdlib.Prelude open;

x : Nat;
x = 7;
```
gives the error
```
  |
6 | x = 7;
  |    ^
expecting :
```

### LINKED ISSUE #2168: The `terminating` keyword is accepted before a function clause
The following type-checks:
```
module term;

import Stdlib.Prelude open;

x : Nat;
terminating
x := 7;
```

### PR DESCRIPTION
**Pull request by lukaszcz**
* Closes #2160 
* Closes #2167 
* Closes #2168

### CODE REVIEW

File: src/Juvix/Compiler/Concrete/Translation/FromParsed/Analysis/Scoping/Error/Types.hs [hunk 1]
```diff
@@ -116,6 +117,42 @@ instance ToGenericError LacksTypeSig where
...
+                        ( over clauseOwnerFunction (set withLocParam x) $
+                            over clausePatterns (PatternAtomIden (NameUnqualified (set withLocParam ":" _clauseOwnerFunction)) :) fc
+                        )
+                    )
+              [x, y] ->
+                line
+                  <> "Perhaps you meant:"
+                  <> line
+                  <> indent'
+                    ( ppCode
```
[the remark this answers is not in the document]
Comment (lukaszcz) [reply src/Juvix/Compiler/Concrete/Translation/FromParsed/Analysis/Scoping/Error/Types.hs:139]: I improved readability here

[the remark this answers is not in the document]
Comment (janmasrovira) [reply src/Juvix/Compiler/Concrete/Translation/FromParsed/Analysis/Scoping/Error/Types.hs:139]: there is code duplication. Everything is the same in both branches except `[":"]` / `[":", y]`

[the remark this answers is not in the document]
Comment (lukaszcz) [reply src/Juvix/Compiler/Concrete/Translation/FromParsed/Analysis/Scoping/Error/Types.hs:139]: Well, yes, you're right. I can move the common part into the function as well.

[the remark this answers is not in the document]
Comment (lukaszcz) [reply src/Juvix/Compiler/Concrete/Translation/FromParsed/Analysis/Scoping/Error/Types.hs:135]: Done

**Review by janmasrovira — APPROVED** [review]

**Pull request merged**